packages/node/src/instance.ts

```diff
@@ -0,0 +1,43 @@
+import i18next, { i18n, InitOptions, Module, Newable, NewableModule, TOptionsBase } from 'i18next';
+
+interface ExtendedI18n extends i18n {
+    getTranslation(lang: string): (key: string, options?: TOptionsBase & Record<string, unknown>) => string
+}
+import FsBackend, { FsBackendOptions } from 'i18next-fs-backend';
+import { getTranslation } from './t';
+
+interface InstanceProps {
+    modules?: Array<Module | NewableModule<Module> | Newable<Module>>;
+    initOptions?: InitOptions;
+}
+
+export function initI18NInstance(options: InstanceProps = {}) {
+    let instance = i18next;
+
+    // Apply FsBackend by default
+    instance = instance.use(FsBackend);
+
+    // Apply additional modules if provided
+    if (options.modules && Array.isArray(options.modules)) {
+        for (const m of options.modules) {
+            instance = instance.use(m);
+        }
+    }
+
+    // Initialize i18next with default and custom options
+    instance.init<FsBackendOptions>({
+        ns: ['translation'],
+        defaultNS: 'translation',
+        initImmediate: false,
+        backend: {
+            loadPath: 'assets/locales/{{lng}}/{{ns}}.json',
+        },
+        ...options.initOptions,
+    });
+
+    // Add getTranslation helper function
+    (instance as ExtendedI18n).getTranslation = getTranslation
```

Comment:
its looks not cool because you are try to modify origin instance.

its will make use not easy to understand because its mix two different logic.

How about split those design? for example, just return origin i18n instance and another `getTranslation` function? make everything is pure and clear.

```
return {
   i18n: instance,
   getTranslation,
}
```

in most case, use dont need know and use i18n. they just only need a `getTranslation` function

the root case is `i18next` manage `instance` global. so i dont wanna have some implicit change which not easy to found and easy to debug

Reply:
Yes, i have thought of what you are suggesting, i used this design for a specific reason. If you look at the code in the example, i initialize the i18n client with some required options, the initialization is required for the code to work. However, if i dont embed the translation function into the initialization, the user might forget to initialize the instance and the translate will not work. Another option may be to initialize when the app is starting, for example, like putting it in the app.ts entrypoint. Please le me know what you think.

Reply:
Yeah, my idea is init and getTranslation function should have not so much relation. its run in different stage.

The only reason is i dont wanna change origin instance, It is somewhat intrusive, and we need to minimize that to ensure our behavior are clear.

for example case, i dont think we have any reason need to inject, because in your example, the only benfit is make sure load timing is right.

but if you change to two different, its will also work fine.

![image](https://github.com/user-attachments/assets/99fb427e-081e-492d-bf58-553f28056381)

Reply:
This is a great solution, thanks.

Reply:
I have fixed it, please take a look.